bottles/frontend/utils/threading.py

```diff
@@ -41,7 +41,7 @@ def __init__(self, task_func, callback=None, *args, **kwargs):
             faulthandler.enable()
 
         self.source_id = None
-        assert threading.current_thread() is threading.main_thread()
+        #assert threading.current_thread() is threading.main_thread()
```

Comment:
If you need to comment this, then something wrong is happening. This line raise an exception when threads are colliding

Reply:
My understanding is that this line forbids to create a thread inside another thread. Is this what's intended? If yes, may I know why it is needed?

Reply:
My first thought that this was to avoid threads launching threads in the case that the parent threads ends before the child, killing it.

But I tested, and even if the parent doesn't join and ends early, the child isn't killed, so I don't think this line is useful.

Reply:
> My understanding is that this line forbids to create a thread inside another thread. Is this what's intended? If yes, may I know why it is needed?

Because Gtk is not thread safe. This line prevents you from accessing the main thread from a second one.

Reply:
I don't really understand how this line prevents that? It just blocks threads to be created by another thread. Unless GTK is run in it's own thread, and threads created from the GTK thread can access it?
If this is why, I feel like this is not a really good solution as it blocks a lot of simplicity of creating threads from the GTK thread, even if those children don't do anything related to GTK (like here for example).

I don't even know how the GTK thread could then create threads from the main thread without great refactoring of the whole architecture.